electron/src/trusted-main-electron-process/SeedableHardwareKeys/IpcServer.ts

```diff
@@ -0,0 +1,100 @@
+import {Server, Socket} from "net";
+import * as net from "net"
+import * as path from "path"
+import * as sudo from "sudo-prompt"
+import * as child_process from 'child_process'
+import fs from "fs";
+import {
+    DeviceListUpdateCallback, ErrorCallback,
+    StopMonitoringFunction
+} from "./UsbDeviceMonitor";
+import {ExecException} from "child_process";
+import {DeviceUniqueIdentifier} from "../../../../common/IElectronBridge";
+import {isWin} from "../../usb";
+
+export var UsbSocketPath : string
+
+if(process.platform == "win32"){
+    UsbSocketPath = path.join('\\\\?\\pipe', process.cwd(), 'dicekeys')
+} else {
+    UsbSocketPath = 'usb.sock'
+}
+
+function escapeParamCmd(value: any): string {
+    // Make sure it's a string
+    // Escape " -> \"
+    // Surround with double quotes
+    return `"${String(value).replace(/"/g, '\\"')}"`;
+}
+let ipcServer: Server | null
+let clientSocket: Socket | null
+export function createIpcServer(deviceListUpdateCallback: DeviceListUpdateCallback, errorCallback?: ErrorCallback) : StopMonitoringFunction{
+    ipcServer = net.createServer((socket: Socket) => {
+        clientSocket = socket
+
+        socket.on('data', (data => {
+            let json = JSON.parse(data.toString())
+
+            if (json.error) {
+                errorCallback?.(json.error)
+            } else if (json.devices) {
+                deviceListUpdateCallback(json.devices)
+            }
+        }))
+    });
+
+    if(!isWin){
+        // clear socket file
+        if(fs.existsSync(UsbSocketPath))  fs.unlinkSync(UsbSocketPath)
+    }
+
+    console.log("Starting IPC Server", UsbSocketPath)
```

Comment:
For situations where we call dicekeys from the command line, it may actually be helpful to have a console message here on windows explaining why the user is about to get a UAC elevation request.